backend/de.metas.business/src/main/java/org/adempiere/mm/attributes/api/impl/SerialNoBL.java

```diff
@@ -53,4 +55,13 @@
 				? Optional.of(serialNo)
 				: Optional.empty();
 	}
+
+	@Override
+	public Optional<AttributeId> getSerialNoAttributeId()
```

Comment:
not sure if this method will be needed after doing other suggested changes

Reply:
still needed.